deltacat/examples/iceberg/iceberg_bucket_writer.py

```diff
@@ -49,11 +72,13 @@ def run(warehouse="s3://my-bucket/my/key/prefix", **kwargs):
                 name="example-iceberg-catalog",
                 # for additional properties see:
                 # https://py.iceberg.apache.org/configuration/
-                properties={
-                    "type": "glue",
-                    "region_name": "us-east-1",
-                    "warehouse": warehouse,
-                },
+                config=IcebergCatalogConfig(
+                    type=CatalogType.GLUE,
+                    properties={
+                        "warehouse": warehouse,
+                        "region_name": "us-east-1",
+                    },
```

Comment:
Seeing this in action like this makes me wonder - do we foresee any benefit from having our own separate IcebergCatalogConfig class? If so, are these internal DeltaCAT developer benefits, or user-facing benefits?

The user-facing initialization here has only become more nested, and potentially less straightforward to understand since it's not a straight passthrough to the existing PyIceberg Catalog properties. Would it be more straightforward to just pass the parameters straight through to the Iceberg catalog like before, so that we don't risk continuous extension of our config class for feature parity going forward?

Reply:
Synced offline - Patrick suggests we should move all config loading from config file stored in disk instead of dynamically set them in code. Tracking issue: https://github.com/ray-project/deltacat/issues/513 .